sdk/identity/azure_identity/src/azure_developer_cli_credential.rs

```diff
@@ -0,0 +1,249 @@
+}
+
+/// Authenticates the identity logged in to the [Azure Developer CLI](https://learn.microsoft.com/azure/developer/azure-developer-cli/overview).
+#[derive(Debug)]
+pub struct AzureDeveloperCliCredential {
+    env: Env,
+    executor: Arc<dyn Executor>,
+    tenant_id: Option<String>,
+}
+
+/// Options for constructing an [`AzureDeveloperCliCredential`].
+#[derive(Clone, Debug, Default)]
+pub struct AzureDeveloperCliCredentialOptions {
+    /// An implementation of [`Executor`] to run commands asynchronously.
+    ///
+    /// If `None`, one is created using [`new_executor`]; alternatively,
+    /// you can supply your own implementation using a different asynchronous runtime.
+    pub executor: Option<Arc<dyn Executor>>,
+
+    /// Identifies the tenant the credential should authenticate in.
+    ///
+    /// Defaults to the azd environment, which is the tenant of the selected Azure subscription.
+    pub tenant_id: Option<String>,
+
+    env: Option<Env>,
+}
+
+impl AzureDeveloperCliCredential {
+    /// Create a new [`AzureDeveloperCliCredential`].
+    pub fn new(
+        options: Option<AzureDeveloperCliCredentialOptions>,
+    ) -> azure_core::Result<Arc<Self>> {
+        let options = options.unwrap_or_default();
+        if let Some(ref tenant_id) = options.tenant_id {
+            validate_tenant_id(tenant_id)?;
+        }
+        let env = options.env.unwrap_or_default();
+        let executor = options.executor.unwrap_or(new_executor());
+        Ok(Arc::new(Self {
+            env,
+            executor,
+            tenant_id: options.tenant_id,
+        }))
+    }
+}
+
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl TokenCredential for AzureDeveloperCliCredential {
+    async fn get_token(&self, scopes: &[&str]) -> azure_core::Result<AccessToken> {
+        if scopes.is_empty() {
+            return Err(Error::new(
+                ErrorKind::Credential,
+                "at least one scope required",
+            ));
+        }
+        let mut command = "azd auth token -o json".to_string();
+        for scope in scopes {
+            validate_scope(scope)?;
+            command.push_str(" --scope ");
+            command.push_str(scope);
+        }
+        if let Some(ref tenant_id) = self.tenant_id {
+            command.push_str(" --tenant-id ");
+            command.push_str(tenant_id);
+        }
+        let (workdir, program, c_switch) = if cfg!(target_os = "windows") {
+            let system_root = self.env.var("SYSTEMROOT").map_err(|_| {
+                Error::message(
+                    ErrorKind::Credential,
+                    "SYSTEMROOT environment variable not set",
+                )
+            })?;
+            (system_root, "cmd", "/C")
+        } else {
+            ("/bin".to_string(), "/bin/sh", "-c")
+        };
+        let command_string = format!("cd {workdir} && {command}");
```

Comment:
Faster to set the working directory. Do we not expose that currently for the executor? Maybe we should.

Reply:
We don't. I could go either way on whether we should. It's not required for any scenario I know about. I expect azure_identity will be the only crate that execs procs in production (we have one Go module that does so only in its tests) and it will always do so via shell like this. I'm happy with the status quo until we find a scenario that requires setting the workdir directly.

If you're curious--I was--I benchmarked `cd` vs setting the workdir in Go just to get an idea of the magnitude of the perf difference. So, here's some trivial anecdata for your Thursday morning: `cd` was faster by up to a few hundred μs and allocated less. I guess setting the workdir requires a larger data structure and/or an additional syscall (I'm not familiar with the kernel API).

Reply:
Interesting. I was thinking of the time the shell would need to parse the extra string/command, execute it, check status (since `&&` was used), and run the next command. May as well leave it, then. If we see a need for setting the PWD elsewhere, we can revisit but I agree we'll probably only need this for `azure_identity`.

Actually, that raises another thought, but I'll start a side thread with you and @LarryOsterman. Not a blocker for this PR.